### PR TITLE
feat: add Typescript type definitions

### PR DESCRIPTION
**Pull request by HashimotoYT**
Added Typescript type definitions for easy use this in Electron projects written on Typescript language.

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+// Definitions by: HashimotoYT <hashimoto.stream@gmail.com>
+
+declare function updater(
+    opts?: {
+        repo?: string,
+        host?: string,
+        updateInterval?: string,
+        logger?: Object
+    }
+): any
```
Comment (MarshallOfSound) [inline index.d.ts]: The updater should return `void`

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+// Type definitions for update-electron-app 1.1.2
+// Project: https://github.com/electron/update-electron-app
+// Definitions by: HashimotoYT <hashimoto.stream@gmail.com>
+
+declare function updater(
+    opts?: {
+        repo?: string,
```
Comment (MarshallOfSound) [inline index.d.ts]: `,` should probably be `;` in type declarations 👍

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+        host?: string,
+        updateInterval?: string,
+        logger?: Object
```
Comment (MarshallOfSound) [inline index.d.ts]: Can we declare the minimal interface for `logger` we expect here.  E.g.

```js
interface ILogger {
  log(message: string): void;
}
```

**Comment by HashimotoYT** [comment]
Ready for another review @MarshallOfSound

**Comment by zeke** [comment]
What do you think about `tslint` or `tsc` (which is it?) as a devDependency and adding it to `npm test` so we have an automated way of checking that the declaration file is valid?

**Comment by MarshallOfSound** [comment]
@zeke There isn't really an easy way to test if the declaration is valid, like the way we do things on `electron/electron` is super non-standard because of how we generate the file.  This file was written by hand so should be assumed to be accurate.

The only way to be 100% sure the declaration is correct is to write the module itself in typescript and have the typescript compiler spit out the declaration

**Comment by zeke** [comment]
I guess I'm thinking more of something like a linter. Not to check that the API declarations are correct, but that the file is at least free of syntax errors.

**Comment by zeke** [comment]
Ah didn't know about `dtslint`. That looks like the right thing. Feel free to add a custom `tslint.json` to this PR if needed.

**Comment by HashimotoYT** [comment]
In last commit I'm add TSLint in recommended configuration, it check only TypeScript declaration file for syntax errors. For check lint in the declaration file you can run `npm run lint`, or run `npm test` for run tests and lint checking.

cc @zeke

**Comment by zeke** [comment]
Great work, @HashimotoYT. Thank you!

**Pull request merged**